File: problem-set-1-mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // Prompt the user for the height of the pyramid
    do
    {
        height = get_int("Height: ");
    }
    while (height <= 0); // Validate user input


    // Build the pyramid
    for (int i = 0; i < height; i++)
    {
        // Print spaces for alignment
        for (int j = height - 1; j > 1; j--)
        {
            printf(" ");
        }

        // Print hashes for the pyramid
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }
      printf("\n"); // Move to the next line for the next

        ////
    }

    return 0;
}

File: problem-set-6-sentimental-mario-less/mario.py
# Import the cs50 library for user input
from cs50 import get_int

# Function to check if the input is a valid integer
def is_valid_integer(n):
    return n > 0 and n <= 8

# Get user input for height
while True:
    height = get_int("Height: ")
    if is_valid_integer(height):
        break
    else:
        print("Invalid input. Please enter a positive integer between 1 and 8 inclusive.")

# Print the pyramid
for i in range(height):
    # Print spaces before the hashes
    print(" " * (height - i - 1), end="")

    # Print the hashes
    print("#" * (i + 1))
